README.md

```diff
@@ -1,3 +1,36 @@
+# Developer Guide
+
+Follow these steps to setup and test the project:
+
+1. **Clone the Repository**  
+   Clone this repository to your local machine:
+
+```bash
+git clone <https://github.com/DistCompiler/dcal.git>
+```
+
+2. **Install Scala CLI:**
+   Install Scala CLI using the using this [link](https://scala-cli.virtuslab.org/install/)
+3. **Install Plugin:**
+   Install the Scala(Metals) plugin by Scalameta in VSCode.
```

Comment:
add `, or equivalent`, so people who don't happen to use VSCode don't feel ignored? (tiny phrasing nit)